lib/collection/src/hash_ring.rs

```diff
@@ -71,6 +94,22 @@ impl<T: Hash + Copy> HashRing<T> {
     }
 }
 
+// `hashring::HashRing` does not implement `Clone`, so we have to do this... 🤦‍♀️
```

Comment:
No reason, other than "this use-case was not considered". E.g., https://github.com/jeromefroe/hashring-rs/pull/20.

Reply:
> Also, can't we take a reference instead of having to do `mem` hacks?

As far as I can tell, no. There is no way to access nodes added into `HashRing`, other than *consume* it into iterator. There's no `fn iter(&self)` or something. Only `impl IntoIterator for HashRing`.

And there's also no `Clone`, so I can't *clone* it and then call `into_iter` on the clone.

Reply:
I see the user in the linked PR having the same problem.

I've added a comment there to hopefully reinstate it.